### PR TITLE
Add client span without context health metric

### PR DESCRIPTION
**Pull request by PerfectSlayer**
# What Does This Do

Add a new health metric to track client (span type) spans without context (ie without parent span).

# Motivation

Have a better understanding of incomplete traces.

# Additional Notes

I also included a small refactoring commit for readability (one method extraction). I can remove it if needed.

### CODE REVIEW

**Review by bantonsson — COMMENTED** [review]
Looks good. Only one small comment.

File: dd-trace-core/src/main/java/datadog/trace/core/monitor/TracerHealthMetrics.java [hunk 1]
```diff
@@ -125,6 +130,19 @@ public void onPublish(final List<DDSpan> trace, final int samplingPriority) {
...
     enqueuedSpans.inc(trace.size());
+    checkForClientSpansWithoutContext(trace);
+  }
+
+  private void checkForClientSpansWithoutContext(final List<DDSpan> trace) {
+    for (DDSpan span : trace) {
+      if (span != null) {
+        long parentId = span.getParentId();
+        String tag = span.getTag(SPAN_KIND, "undefined");
+        if (parentId == ZERO && SPAN_KIND_CLIENT.equals(tag)) {
```
Comment (bantonsson) [inline dd-trace-core/src/main/java/datadog/trace/core/monitor/TracerHealthMetrics.java:141]: I would move the `span.getTag(...)` to after `parentId == ZERO` to only do it when necessary.

**Review by bantonsson — COMMENTED** [review]
And now with the comment...

File: dd-trace-core/src/main/java/datadog/trace/core/monitor/TracerHealthMetrics.java (hunk 1 above)
Comment (PerfectSlayer) [reply dd-trace-core/src/main/java/datadog/trace/core/monitor/TracerHealthMetrics.java]: Agreed. I patched it and will squash the commits.

**Pull request merged**